core/src/sampling/RandomSampler.cpp

```diff
@@ -139,69 +139,27 @@ void RandomSampler::draw(std::vector<size_t>& result,
                          size_t max,
                          const std::set<size_t>& skip,
                          size_t num_samples) {
-  if (num_samples < max / 2) {
-    draw_simple(result, max, skip, num_samples);
-  } else {
-    draw_knuth(result, max, skip, num_samples);
-  }
-}
 
-void RandomSampler::draw_simple(std::vector<size_t>& result,
-                                size_t max,
-                                const std::set<size_t>& skip,
-                                size_t num_samples) {
-  result.reserve(num_samples);
+  // Create indices
+  result.resize(max);
```

Comment:
For the logic in this method, I also think we should also go with exactly what ranger has. The ranger maintainers are generally very good about benchmarking changes, so we'll have confidence that the approach is fast. It also makes it easier to debug our code when there is a direct comparison to ranger when possible, as we can consult their bug reports, PRs, etc.

Reply:
Sure -- I was a bit thrown off by `grf`s use of a `std::set` instead of a `std::vector` for `skip`, but I've figured it out.

Reply:
Makes sense, I had actually forgotten this was the case (and had to change my comment below to read 'log(n)' :) ).